src/StreetNameRegistry/Municipality/Municipality.cs

```diff
@@ -108,15 +108,29 @@ public void CorrectStreetNameName(Names streetNameNames, PersistentLocalId persi
             streetName.CorrectNames(streetNameNames, GuardStreetNameNames);
         }
 
+        public void CorrectStreetNameRetirement(Names streetNameNames, PersistentLocalId persistentLocalId)
+        {
+            var streetName = StreetNames.FindByPersistentLocalId(persistentLocalId);
+
+            if (streetName is null)
+            {
+                throw new StreetNameIsNotFoundException(persistentLocalId);
+            }
+
+            if (MunicipalityStatus != MunicipalityStatus.Current)
+            {
+                throw new MunicipalityHasInvalidStatusException();
+            }
+
+            streetName.CorrectRetirement(() => GuardUniqueActiveStreetNameNames(streetNameNames, persistentLocalId));
```

Comment:
Use `streetName.Names` instead of `streetNameNames` when calling `GuardUniqueActiveStreetNameNames`.